_posts/fr/2022-06-30-analyse-cout-electricite.md

```diff
@@ -0,0 +1,304 @@
+Du 2021-04-12T00:00:00+01:00 au;AA000000;Option Heures Creuses + Week-End;HC;Heures Creuses;EAS F1;HP;Heures Pleines;
+```
+
+Les données de consommation ont une structure spécifique. Les deux premières lignes sont des en-tête qui décrivent la
+nature
+des données. Un élément doit attirer notre attention : la consommation est en watt-heure et c'est du cumulatif
+(comme un compteur kilométrique d'une voiture).
+
+La troisième ligne ce sont les noms de colonnes. Ces noms sont des index de colonnes. Les labels plus détails sont dans
+les 2 dernières lignes du tableau.
+
+Par exemple, pour la colonne `EAS F1` j'ai la valeur `11538434`. En regardant les deux dernière lignes du tableau, je
+vois
+
+```text
+Identifiant classe temporelle 1;Libelle classe temporelle 1;Cadran classe temporelle 1
+EAS F1                         ;HP                         ;Heures Pleines
+```
+
+Donc la colonne `EAS F1` correspond à la consommation en `Heures Pleines`.
+
+Au final, ce sont les colonnes `EAS F1`, `EAS F2`, `EAS F3` qui correspondent à la consommation en heures pleines,
+heures creuses et weekend.
+
+Dans un premier temps je supprimes les lignes qui ne m'intéressent pas.
+
+```python
+BASE_PATH = "datalake/"
+RAW_DATA_FILENAME = "sample.csv"
+RAW_DATA_CLEANED_FILENAME = "enedis_conso_clean.csv"
+CLEANED_DATA_FILENAME = "enedis-data-ready-to-work.parquet"
+
+with open(f"{BASE_PATH}{RAW_DATA_FILENAME}", "r+") as fp:
+    lines = fp.readlines()
+    with open(f"{BASE_PATH}{RAW_DATA_CLEANED_FILENAME}", "w+") as fpclean:
+        fpclean.writelines(lines[2: len(lines) - 2])
+```
+
+Je relis le fichier avec Pandas. Je vais effectuer quelques calcul de bases comme le différentiel par jour et la
+consommation totale de la journée.
+
+Avec Pandas, le travail sur les données s'effectue en colonne et non en ligne. Ce fonctionnement est un peu déroutant
+lorsque l'on utilise
+courament des bases de données tel que MySQL ou PostgreSQL.
+
+```python
+import pandas as pd
+
+df = pd.read_csv(f"{BASE_PATH}{RAW_DATA_CLEANED_FILENAME}", sep=";")
+
+# Calcul de la consommation totale de la journée
+df["Total Consommation"] = df["EAS F2"] + df["EAS F1"] + df["EAS F3"]
+
+# Renommage des colonnes
+df_work = df[["Horodate", "EAS F1", "EAS F2", "EAS F3", "Total Consommation"]].copy()
+df_work.columns = [
+    "Horodate",
+    "Quantité Heure Creuse en wh",
+    "Quantité Heure Pleine en wh",
+    "Quantité Weekend en wh",
+    "Total Consommation",
+]
+
+# Calcul de la différence par rapport à la journée précédente pour chaque colonne.
+df_work["Différence journalier Total Consommation"] = df_work["Total Consommation"].diff()
+df_work["Différence journalier Heure Creuse"] = df_work["Quantité Heure Creuse en wh"].diff()
+df_work["Différence journalier Heure Pleine"] = df_work["Quantité Heure Pleine en wh"].diff()
+df_work["Différence journalier Weekend"] = df_work["Quantité Weekend en wh"].diff()
```

Comment:
les `df.diff()`, retrournent `NaN` pour la première ligne il me semble, c'est pas problématique ?

Reply:
ça ne m'avait pas interpelé au début car ça me semblait normal. la colonne est bien typé en float64.

en cherchant, NaN signifie "valeur manquante" du point de vue de Pandas.
- https://pandas.pydata.org/pandas-docs/stable/user_guide/missing_data.html#values-considered-missing
- https://chartio.com/resources/tutorials/how-to-check-if-any-value-is-nan-in-a-pandas-dataframe/

donc, ça ne pose pas de problème et c'est le fonctionnement normal.

Reply:
@lepiaf Merci pour la précision :)